### DISCUSSION TITLE
Contract deployment is pending

### DISCUSSION THREAD

**Discussion by Moemen12** [Cyfrin/foundry-full-course-cu#2105] [Q&A] [answered]
hello after finishing my first contract and trying to deploy it using inject provider metamask , it's pending , idk it's like take too long time.
and if i trying to cancel or speed up anything ,nothing happens after submitting , nothing cancelled nothing being changes

![Screenshot 2024-07-25 145601](https://github.com/user-attachments/assets/cef233eb-65c2-4cba-937f-927ab3c4a7fa)

![Screenshot 2024-07-25 145413](https://github.com/user-attachments/assets/fd079398-86fa-4022-96cc-537cd20b344d)

![Screenshot 2024-07-25 145214](https://github.com/user-attachments/assets/392ca99d-0f2c-4c14-90f7-c5524fe9fbff)

**Comment by JoshuyPan** [comment]
Hi @Moemen12,
Try to use "WalletConnection" instead of Injected Provider-Metamask into the deploy tab.

**Reply by Moemen12** [reply]
i have noticed something when i hover over cancel button to cancel it im gettting _no enough gas_

yeah the problem is related to gas .. how to increase it to deploy or what should i do ? i tried wallet connection but i got this error

![Screenshot 2024-07-25 153848](https://github.com/user-attachments/assets/69c4c862-4da6-4bde-bb5f-e7218599e55b)

**Reply by EngrPips** [reply]
This seems like you don't have enough ether to process the deployment. Maybe try to get some sepolia eth chainlink faucet `https://faucets.chain.link/fuji`

**Reply by Moemen12** [reply]
i tried to send some ether but the same result 

![Screenshot 2024-07-25 172329](https://github.com/user-attachments/assets/dcdd1064-0af9-412b-a933-729cee5ff6aa)

**Reply by EngrPips** [reply, marked as answer]
Maybe the blockchain is congested. Let's wait to see if those pending transactions will fail or be successful

**Reply by Moemen12** [reply]
u can here from this link how it's pending

**https://sepolia.etherscan.io/tx/0xda247076405f6611bfd43fbeaab46e05e9d41d6556eca0054fe2f7422c72dc86**


![Screenshot 2024-07-25 180728](https://github.com/user-attachments/assets/3cebc0e9-bb01-4fc7-ab5c-025a93c9a2dd)

**Reply by EngrPips** [reply]
Thanks for sharing the transaction link.

**Reply by Moemen12** [reply]
it has been deployed successfully but it takes to long why? more than 5 hours

**Reply by EngrPips** [reply]
I heard Sepolaia test net has been slow lately because of congestion. Sorry you have to wait that long but I am glad at least you have it deployed now.

**Reply by EngrPips** [reply]
I heard Sepolia test net has been slow lately because of congestion. Sorry you have to wait that long but I am glad at least you have it deployed now.
